api/spec/v1.yaml

```diff
@@ -1638,6 +1670,52 @@ components:
       description: |
         An event emitted by the consensus layer.
 
+    RoothashMessageType:
+      type: string
+      enum:
+        - staking.transfer
+        - staking.withdraw
+        - staking.add_escrow
+        - staking.reclaim_escrow
+        - registry.update_runtime
+        - governance.cast_vote
+        - governance.submit_proposal
+      example: *roothash_message_type_1
+
+    RoothashMessageList:
+      allOf:
+        - $ref: '#/components/schemas/List'
+        - type: object
+          required: [roothash_messages]
+          properties:
+            roothash_messages:
+              type: array
+              items:
+                allOf: [$ref: '#/components/schemas/RoothashMessage']
+
+    RoothashMessage:
+      type: object
+      required: [runtime, round, index]
+      properties:
+        runtime:
+          type: string
+        round:
+          type: integer
+          format: int64
+        index:
+          type: integer
+          format: int32
+        type:
+          allOf: [$ref: '#/components/schemas/RoothashMessageType']
+        body:
+          type: object
```

Comment:
Can we add a pointer to the relevant structures in oasis-core here? So people have a semi-formal way of anticipating the expected data structures, even if we give no strict guarantees about the shape of this `object`. Ditto for `result`.

Reply:
leme see if we have a place that documents these

Reply:
https://pkg.go.dev/github.com/oasisprotocol/oasis-core/go/roothash/api/message#Message ah yeah